File: promomap/src/main/java/br/com/promomap/model/graph/Arris.java
/*
 * @(#)Arris.java 28 de mai de 2017 - 21:57:10
 *
 */
package br.com.promomap.model.graph;

import java.math.BigDecimal;

/**
 * @author <a href="mailto:leandro.lucas_@hotmail.com">Leandro Lucas Santos</a>
 */
public class Arris {

	private Vertex vertexA;

	private Vertex vertexB;

	private BigDecimal distance;

	public Vertex getVertexA() {
		return vertexA;
	}

	public void setVertexA(Vertex vertexA) {
		this.vertexA = vertexA;
	}

	public Vertex getVertexB() {
		return vertexB;
	}

	public void setVertexB(Vertex vertexB) {
		this.vertexB = vertexB;
	}

	public BigDecimal getDistance() {
		return distance;
	}

	public void setDistance(BigDecimal distance) {
		this.distance = distance;
	}

}

File: promomap-web/src/app/models/Arris.ts
import { Vertex } from './Vertex';

export class Arris {

  private distance: number;
  private vertexA: Vertex;
  private vertexB: Vertex;


  public getDistance(): number {
    return this.distance;
  }

  public setDistance(distance: number): void {
    this.distance = distance;
  }

  public getVertexA(): Vertex {
    return this.vertexA;
  }

  public setVertexA(vertexA: Vertex): void {
    this.vertexA = vertexA;
  }

 public getVertexB(): Vertex {
    return this.vertexB;
  }

  public setVertexB(vertexB: Vertex): void {
    this.vertexB = vertexB;
  }
}
